docs/html/conf.py

```diff
@@ -12,22 +12,30 @@
 sys.path.insert(0, docs_dir)
 
 # -- General configuration ------------------------------------------------------------
-
 extensions = [
-    # first-party extensions
-    "sphinx.ext.autodoc",
-    "sphinx.ext.todo",
-    "sphinx.ext.extlinks",
-    "sphinx.ext.intersphinx",
-    # our extensions
+    # extensions common to 'man' and 'html' builds
     "pip_sphinxext",
-    # third-party extensions
-    "myst_parser",
-    "sphinx_copybutton",
-    "sphinx_inline_tabs",
-    "sphinxcontrib.towncrier",
 ]
 
+# 'tags' is a special object handled by sphinx
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-tags
+if "html" in tags or "READTHEDOCS" in os.environ:  # type: ignore[name-defined] # noqa: F821
+    # html specific deps
+    extensions.extend(
```

Comment:
It occurred to me that maybe if the extension loading was to be moved to the `setup()` entry point, the builder name would already be known by then and so this might not even have to use tags but would rely on the requested builders.

Another approach I used in the past, in a little different context, was scanning for the builder args in `sys.argv`: https://github.com/sphinx-contrib/spelling/issues/55#issuecomment-655546944 / https://github.com/cherrypy/cheroot/commit/63097fbfd40eda733a14893a6ed7d1e370a8cde1#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R12-R15.